libraries/kotlin.test/common/src/test/kotlin/kotlin/test/tests/BasicAssertionsTest.kt

```diff
@@ -19,6 +19,29 @@ class BasicAssertionsTest {
         assertEquals("a", "a")
     }
 
+    @Test
+    fun testAssertFailsWithMessage() {
+        assertFailsWith<NoSuchElementException>("This message is asserted") {
+            throw NoSuchElementException("This message is asserted")
+        }
+    }
+
+    @Test
+    fun testAssertFailsWithMessageFails() {
+        assertTrue(true) // at least one assertion required for qunit
+
+        withDefaultAsserter run@ {
+            try {
+                assertFailsWith<IllegalArgumentException>("message") {
+                    throw IllegalArgumentException("another message")
+                }
+            } catch (e: AssertionError) {
+                return@run
```

Comment:
You should assert on the message here actually indicating a mis-matched message. This also obviates the need for the dummy assertTrue.